Miscellaneous printing fixes

mhyee opened the pull request:
Print to ostream parameter, not std::cout.

Also other miscellaneous fixes (see second commit).

---

Before, we would get something like:

```
╞══════════════════════════╡  Compiling bitwShiftL  ╞══════════════════════════╡
bitwiseShiftL
┌──────────────────────────────────────────────────────────────────────────────┐
│ bitwShiftL_0x2a6ecc8                                                         │
├────── Final PIR Version
Closure 0x2792ad0(0x2a6ecc8)
BB0
  val^    %0.0  = LdArg            0
  val^    %0.1  = LdArg            1
  env     e0.2  = MkEnv            a=%0.0, n=%0.1, parent=<environment: namespace:base>
  val     %0.3  = Force            %0.0, e0.2
  val^?   %0.4  = LdVar            n, e0.2
  val     %0.5  = Force            %0.4, e0.2
  val^    %0.6  = CallBuiltin      (%0.3, %0.5) e0.2
  val     %0.7  = Force            %0.6, e0.2
  void            Return           %0.7
```

Note the stray `bitwiseShiftL` at the top, and how `CallBuiltin` is calling some unnamed function.

The correct output is:

```
╞══════════════════════════╡  Compiling bitwShiftL  ╞══════════════════════════╡
┌──────────────────────────────────────────────────────────────────────────────┐
│ bitwShiftL_0x2a6ecc8                                                         │
├────── Final PIR Version
Closure 0x2792ad0(0x2a6ecc8)
BB0
  val^    %0.0  = LdArg            0
  val^    %0.1  = LdArg            1
  env     e0.2  = MkEnv            a=%0.0, n=%0.1, parent=<environment: namespace:base>
  val     %0.3  = Force            %0.0, e0.2
  val^?   %0.4  = LdVar            n, e0.2
  val     %0.5  = Force            %0.4, e0.2
  val^    %0.6  = CallBuiltin      bitwiseShiftL(%0.3, %0.5) e0.2
  val     %0.7  = Force            %0.6, e0.2
  void            Return           %0.7
```

mhyee committed:
Fix printing of CallBuiltin PIR instruction.

Print to ostream parameter, not std::cout.

JanJecmen reviewed (commented):
Good one.
I did a search for `std::cout` and there are a couple of places that are dubious, but maybe they can go into a separate pr...

- runtime.cpp: `printFunction` creates `stringstream` and then dumps it into `cout`, could print directly into `cout`.. Also, it mixes `Rprintf` and `cout`
- type.cpp: the pir type prints always into `cout` (it looks as if it's not used at all though)
- scope.cpp: `TheScopeAnalysis::print` mixes `out` and `cout` printing (hidden behind `if (false)`)

Probably leftovers from debugging? Anyway, if you feel it's not necessary to change these, this PR is fine as is.

mhyee committed:
More printing fixes

- remove unnecessary stringstreams
- print errors to cerr
- print to ostream parameter instead of cout
- remove uses of Rprintf
- fix formatting of printing Code objects

mhyee commented:
I fixed the issues you mentioned, and made a few other ones as well.

I skipped over the things that were more annoying to change, or were only in commented out code.

mhyee commented on rir/src/interpreter/runtime.cpp:
```diff
@@ -28,27 +28,45 @@ Function* isValidClosureSEXP(SEXP closure) {
-    Rprintf("  Signature:       %p\n", f->signature);
-    if (f->signature)
-        f->signature->print();
+void printFunction(Function* f, std::ostream& out) {
```
This actually looks uglier, but the advantage is that we can control where the printing happens, rather than trying to hack Rprintf.

o- commented:
thanks for doing another pass here!
